File: go/solutions/find_closest_person.go
package solutions

func FindClosest(x int, y int, z int) int {
	abs := func(n int) int {
		if n < 0 {
			return -1 * n
		}
		return n
	}
	if abs(x-z) == abs(y-z) {
		return 0
	}
	if abs(x-z) > abs(y-z) {
		return 2
	}
	return 1
}

File: python/src/find_closest_person.py
def findClosest(x: int, y: int, z: int) -> int:
    match (abs(x - z), abs(y - z)):
        case (a, b) if a == b:
            return 0
        case (a, b) if a > b:
            return 2
        case _:
            return 1
